al_bench/dataset.py

```diff
@@ -268,7 +268,7 @@ def set_all_feature_vectors(self, feature_vectors: NDArray) -> None:
         """
         Set the entire database of feature vectors from a supplied numpy array.
         """
-        if isinstance(feature_vectors, np.ndarray) and len(feature_vectors.shape) == 2:
+        if isinstance(feature_vectors, np.ndarray) and len(feature_vectors.shape) >= 2:
```

Comment:
The >=2 is for 2D data + batch dimension?

Reply:
Yes, more or less.  The first dimension has size equal to the number of examples in the data set.  (When training is done, this will be reduced to the batch size.)  If there is 1 more dimension (for a total of 2), it is likely the length of a feature vector.  If there are 2 dimensions beyond the data set size (for a total of 3), the two are likely the height and width of an image or image tile.